diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/melder_audio.cpp demo_praat_sources/sys/melder_audio.cpp
--- praat_sources/sys/melder_audio.cpp	2014-12-29 08:24:04.000000000 +0100
+++ demo_praat_sources/sys/melder_audio.cpp	2015-05-12 09:48:06.000000000 +0200
@@ -515,10 +515,14 @@ void MelderAudio_play16 (int16_t *buffer
 	my numberOfChannels = numberOfChannels;
 	my callback = playCallback;
 	my closure = playClosure;
+	#ifdef DEMOAPP
+		my asynchronicity = kMelder_asynchronicityLevel_ASYNCHRONOUS;
+	#else
 	my asynchronicity =
 		Melder_batch ? kMelder_asynchronicityLevel_SYNCHRONOUS :
 		(Melder_backgrounding && ! Melder_asynchronous) ? kMelder_asynchronicityLevel_INTERRUPTABLE :
 		kMelder_asynchronicityLevel_ASYNCHRONOUS;
+	#endif
 	if (my asynchronicity > preferences. maximumAsynchronicity)
 		my asynchronicity = preferences. maximumAsynchronicity;
 	trace (U"asynchronicity ", my asynchronicity);
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/stat/praat_Stat.cpp demo_praat_sources/stat/praat_Stat.cpp
--- praat_sources/stat/praat_Stat.cpp	2011-10-01 13:32:45.000000000 +0200
+++ demo_praat_sources/stat/praat_Stat.cpp	2011-11-18 08:55:32.000000000 +0100
@@ -420,7 +420,9 @@ DO
 END2 }
 
 DIRECT2 (Table_edit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot edit a Table from batch.");
+#endif
 	LOOP {
 		iam (Table);
		autoTableEditor editor = TableEditor_create (ID_AND_FULL_NAME, me);
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat_script.cpp demo_praat_sources/sys/praat_script.cpp
--- praat_sources/sys/praat_script.cpp	2011-09-02 13:53:33.000000000 +0200
+++ demo_praat_sources/sys/praat_script.cpp	2011-10-28 07:54:25.000000000 +0200
@@ -585,6 +585,7 @@ void praat_executeScriptFromText (wchar 
 		autoInterpreter interpreter = Interpreter_create (NULL, NULL);
 		Interpreter_run (interpreter.peek(), text);
 	} catch (MelderError) {
+		exit (0);
 		Melder_throw (U"Script not completed.");
 	}
 }
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/Graphics_text.cpp demo_praat_sources/sys/Graphics_text.cpp
--- praat_sources/sys/Graphics_text.cpp	2011-09-29 12:27:01.000000000 +0200
+++ demo_praat_sources/sys/Graphics_text.cpp	2011-10-27 09:14:57.000000000 +0200
@@ -146,7 +146,7 @@ static HFONT loadFont (GraphicsScreen me
 		ipaInited = TRUE;
 		if (! charisAvailable && ! doulosAvailable) {
 			/* BUG: The next warning may cause reentry of drawing (on window exposure) and lead to crash. Some code must be non-reentrant !! */
-			Melder_warning (U"The phonetic font is not available.\nSeveral characters may not look correct.\nSee www.praat.org");
+			//Melder_warning (U"The phonetic font is not available.\nSeveral characters may not look correct.\nSee www.praat.org");
 		}
 	}
 	wcscpy (spec. lfFaceName,
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat.cpp demo_praat_sources/sys/praat.cpp
--- praat_sources/sys/praat.cpp	2014-04-22 11:09:57.000000000 +0200
+++ demo_praat_sources/sys/praat.cpp	2014-04-22 11:53:58.000000000 +0200
@@ -1144,7 +1144,7 @@ void praat_init (const char *title, unsi
 		 * Make sure that the program's private directory exists.
 		 */
 		#if defined (UNIX)
-			Melder_sprint (name,256, U".", programName, U"-dir");   // for example .myProg-dir
+			Melder_sprint (name,256, U".", programName, U"");   // for example .myProg-dir
 		#elif defined (macintosh)
 			Melder_sprint (name,256, praatP.title, U" Prefs");   // for example MyProg Prefs
 		#elif defined (_WIN32)
@@ -1217,6 +1217,12 @@ void praat_init (const char *title, unsi
 
 	GuiWindow raam = NULL;
 	if (Melder_batch) {
+		#if defined (DEMOAPP) && gtk
+			char objectWindowTitle [100];
+			Machine_initLookAndFeel (argc, argv);
+			sprintf (objectWindowTitle, "%s Objects", praatP.title);
+			GuiWindow_create (0, 0, WINDOW_WIDTH, WINDOW_HEIGHT, 0, 0, Melder_peek8to32 (objectWindowTitle), gui_cb_quit, NULL, 0);
+		#endif
 		#if defined (UNIX) || defined (macintosh) || defined (_WIN32) && defined (CONSOLE_APPLICATION)
 			MelderString_empty (& theCurrentPraatApplication -> batchName);
 			for (unsigned int i = iarg_batchName; i < argc; i ++) {
@@ -1263,9 +1269,11 @@ void praat_init (const char *title, unsi
 		#if motif
 			GuiApp_setApplicationShell (theCurrentPraatApplication -> topShell -> d_xmShell);
 		#endif
+#ifndef DEMOAPP
 		trace (U"before objects window shows locale ", Melder_peek8to32 (setlocale (LC_ALL, NULL)));
 		GuiThing_show (raam);
 		trace (U"after objects window shows locale ", Melder_peek8to32 (setlocale (LC_ALL, NULL)));
+#endif
 	}
 	Thing_recognizeClassesByName (classCollection, classStrings, classManPages, classSortedSetOfString, NULL);
 	if (Melder_batch) {
@@ -1275,6 +1283,9 @@ void praat_init (const char *title, unsi
 		trace (U"adding fixed buttons without GUI");
 		praat_addFixedButtons (NULL);
 	} else {
+		#ifdef DEMOAPP
+			Melder_backgrounding = true;
+		#endif
 
 		#ifdef macintosh
 			#if ! useCarbon
@@ -1302,7 +1313,9 @@ void praat_init (const char *title, unsi
 		trace (U"creating the dynamic menu in the Objects window");
 		praat_actions_createDynamicMenu (raam);
 		trace (U"showing the Objects window");
+#ifndef DEMOAPP
 		GuiThing_show (raam);
+#endif
 	//Melder_fatal (U"stop");
 		#if defined (UNIX) && ! defined (NO_GRAPHICS)
 			try {
@@ -1331,7 +1344,12 @@ void praat_init (const char *title, unsi
 
 	trace (U"creating the Picture window");
 	trace (U"before picture window shows: locale is ", Melder_peek8to32 (setlocale (LC_ALL, NULL)));
-	if (! praatP.dontUsePictureWindow) praat_picture_init ();
+	#ifdef DEMOAPP
+		praat_picture_init ();
+		praat_dontUsePictureWindow ();
+	#else
+		if (! praatP.dontUsePictureWindow) praat_picture_init ();
+	#endif
 	trace (U"after picture window shows: locale is ", Melder_peek8to32 (setlocale (LC_ALL, NULL)));
 }
 
@@ -1480,11 +1498,11 @@ void praat_run (void) {
 		if (! str32equ (Melder_integer (dummy), U"1000000000000"))
 			Melder_fatal (U"The number 1000000000000 is mistaken written on this machine as ", dummy, U".");
 	}
-	{ uint32_t dummy = 0xffffffff;
-		Melder_assert ((int64) dummy == 4294967295LL);
-		Melder_assert (str32equ (Melder_integer (dummy), U"4294967295"));
-		Melder_assert (double (dummy) == 4294967295.0);
-	}
+//	{ uint32_t dummy = 0xffffffff;
+//		Melder_assert ((int64_t) dummy == 4294967295LL);
+//		Melder_assert (str32equ (Melder_integer (dummy), U"4294967295"));
+//		Melder_assert (double (dummy) == 4294967295.0);
+//	}
 	{ double dummy = 3000000000.0;
 		Melder_assert ((uint32) dummy == 3000000000);
 	}
@@ -1628,6 +1646,14 @@ void praat_run (void) {
 						s ++;
 					}
 				}
+				else if (thePraatStandAloneScriptText != NULL) {
+					try {
+						praat_executeScriptFromText (thePraatStandAloneScriptText);
+						praat_exit (0);
+					} catch (MelderError) {
+						Melder_flushError (NULL);
+					}
+				};
 			#endif
 			for (;;) {
 				XEvent event;
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat_picture.cpp demo_praat_sources/sys/praat_picture.cpp
--- praat_sources/sys/praat_picture.cpp	2012-09-27 09:10:50.000000000 +0200
+++ demo_praat_sources/sys/praat_picture.cpp	2012-10-15 13:47:43.000000000 +0200
@@ -1511,8 +1511,10 @@ void praat_picture_open (void) {
 		#elif cocoa
 			GuiThing_show (dialog);
 		#elif motif
+#ifndef DEMOAPP
 			XtMapWidget (dialog -> d_xmShell);
 			XMapRaised (XtDisplay (dialog -> d_xmShell), XtWindow (dialog -> d_xmShell));
+#endif
 		#endif
 		Picture_unhighlight (praat_picture);
 	}
@@ -1796,7 +1798,9 @@ void praat_picture_init (void) {
 		width = height = resolution * 12;
 		scrollWindow = GuiScrolledWindow_createShown (dialog, margin, 0, Machine_getMenuBarHeight () + margin, 0, 1, 1, 0);
 		drawingArea = GuiDrawingArea_createShown (scrollWindow, width, height, NULL, NULL, NULL, NULL, NULL, 0);
+#ifndef DEMOAPP
 		GuiThing_show (dialog);
+#endif
 	}
 
 	// TODO: Paul: deze moet VOOR de update functies anders krijgen die void_me 0x0
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/fon/praat_TextGrid_init.cpp demo_praat_sources/fon/praat_TextGrid_init.cpp
--- praat_sources/fon/praat_TextGrid_init.cpp	2011-09-12 10:17:12.000000000 +0200
+++ demo_praat_sources/fon/praat_TextGrid_init.cpp	2011-09-12 10:59:25.000000000 +0200
@@ -855,7 +855,9 @@ static void cb_TextGridEditor_publicatio
 	}
 }
 DIRECT2 (TextGrid_edit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	Sound sound = NULL;
 	LOOP {
 		if (CLASS == classSound) sound = (Sound) OBJECT;   // may stay NULL
@@ -886,7 +888,9 @@ DIRECT (TextGrid_edit)
 END2 }
 
 DIRECT2 (TextGrid_LongSound_edit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	LongSound longSound = NULL;
 	int ilongSound = 0;
 	LOOP {
@@ -902,7 +906,9 @@ DIRECT (TextGrid_LongSound_edit)
 END2 }
 
 DIRECT2 (TextGrid_SpellingChecker_edit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	SpellingChecker spellingChecker = NULL;
 	int ispellingChecker = 0;
 	Sound sound = NULL;
@@ -919,7 +925,9 @@ DIRECT (TextGrid_SpellingChecker_edit)
 END2 }
 
 DIRECT2 (TextGrid_LongSound_SpellingChecker_edit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	LongSound longSound = NULL;
 	SpellingChecker spellingChecker = NULL;
 	int ilongSound = 0, ispellingChecker = 0;
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/fon/praat_Sound_init.cpp demo_praat_sources/fon/praat_Sound_init.cpp
--- praat_sources/fon/praat_Sound_init.cpp	2012-10-11 10:39:47.000000000 +0200
+++ demo_praat_sources/fon/praat_Sound_init.cpp	2012-10-18 15:09:22.000000000 +0200
@@ -172,7 +172,9 @@ DO
 END2 }
 
 DIRECT2 (LongSound_view) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a LongSound from batch.");
+#endif
 	LOOP {
 		iam (LongSound);
 		autoSoundEditor editor = SoundEditor_create (ID_AND_FULL_NAME, me);
@@ -717,7 +719,9 @@ static void cb_SoundEditor_publication (
 	}
 }
 DIRECT2 (Sound_edit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a Sound from batch.");
+#endif
 	LOOP {
 		iam (Sound);
 		autoSoundEditor editor = SoundEditor_create (ID_AND_FULL_NAME, me);
@@ -1396,8 +1400,10 @@ static void cb_SoundRecorder_publication
 	praat_updateSelection ();
 }
 static void do_Sound_record (int numberOfChannels) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch)
 		Melder_throw (U"Cannot record a Sound from batch.");
+#endif
 	if (theSoundRecorder) {
 		if (numberOfChannels == thePreviousNumberOfChannels) {
 			Editor_raise (theSoundRecorder);
@@ -1447,10 +1453,21 @@ FORM (Sound_recordFixedTime, U"Record So
 	POSITIVE (U"Duration (seconds)", U"1.0")
 	OK2
 DO
+#if defined (_WIN32) && defined (DEMOAPP)
+	UINT numberOfDevices = waveInGetNumDevs (), i;
+	if(numberOfDevices > 0) {
+#endif
 	autoSound me = Sound_recordFixedTime (GET_INTEGER (U"Input source"),
 		GET_REAL (U"Gain"), GET_REAL (U"Balance"),
 		Melder_atof (GET_STRING (U"Sampling frequency")), GET_REAL (U"Duration"));
 	praat_new (me.transfer(), U"untitled");
+#if defined (_WIN32) && defined (DEMOAPP)
+	} else {
+	// Create an empty sound if no input is available
+	autoSound me = Sound_createSimple (1, GET_REAL (U"Duration"), GET_REAL (U"Sampling frequency"));
+	praat_new (me.transfer(), U"untitled");	
+	};
+#endif
 END2 }
 
 FORM (Sound_resample, U"Sound: Resample", U"Sound: Resample...") {
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/Ui.cpp demo_praat_sources/sys/Ui.cpp
--- praat_sources/sys/Ui.cpp	2014-01-27 09:02:44.000000000 +0100
+++ demo_praat_sources/sys/Ui.cpp	2014-03-27 12:22:58.000000000 +0100
@@ -765,12 +765,16 @@ void UiForm_finish (I) {
 	iam (UiForm);
 	if (! my d_dialogParent && ! my isPauseForm) return;
 
+	int numberOfContinueButtons = my numberOfContinueButtons;
 	int size = my numberOfFields;
 	int dialogHeight = 0, x = Gui_LEFT_DIALOG_SPACING, y;
 	int textFieldHeight = Gui_TEXTFIELD_HEIGHT;
-	int dialogWidth = 520, dialogCentre = dialogWidth / 2, fieldX = dialogCentre + Gui_LABEL_SPACING / 2;
+	int dialogWidth = REVERT_BUTTON_WIDTH + STOP_BUTTON_WIDTH + 2*Gui_LABEL_SPACING + numberOfContinueButtons*(STANDARDS_BUTTON_WIDTH + Gui_LABEL_SPACING), dialogCentre = dialogWidth / 2, fieldX = dialogCentre + Gui_LABEL_SPACING / 2;
 	int labelWidth = fieldX - Gui_LABEL_SPACING - x, fieldWidth = labelWidth, halfFieldWidth = fieldWidth / 2 - 6;
 
+	/* Ensure that the dialogs are wide enough */
+	if (dialogWidth<520) dialogWidth = 520;
+	
 	GuiForm form;
 
 	/*
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/UiPause.cpp demo_praat_sources/sys/UiPause.cpp
--- praat_sources/sys/UiPause.cpp	2012-09-27 09:10:50.000000000 +0200
+++ demo_praat_sources/sys/UiPause.cpp	2012-09-27 13:25:18.000000000 +0200
@@ -156,6 +156,7 @@ int UiPause_end (int numberOfContinueBut
 		{// scope
 			autoMelderSaveDefaultDir saveDir;
 			thePauseForm_clicked = 0;
+			if(str32nequ(continueText1, U"DESTROY WINDOW ", 15))thePauseForm_clicked = 1;
 			Melder_assert (theEventLoopDepth == 0);
 			theEventLoopDepth ++;
 			try {
@@ -179,24 +180,28 @@ int UiPause_end (int numberOfContinueBut
 						[pool release];
 					} while (! thePauseForm_clicked);
 				#elif motif
+					if(str32nequ(continueText1, U"DESTROY WINDOW ", 15) == 0){
 					do {
 						XEvent event;
 						GuiNextEvent (& event);
 						XtDispatchEvent (& event);
 					} while (! thePauseForm_clicked);
+					};
 				#endif
 			} catch (MelderError) {
 				Melder_flushError (U"An error made it to the outer level in a pause window; should not occur! Please write to paul.boersma@uva.nl");
 			}
 			theEventLoopDepth --;
 		}
+		if(str32nequ(continueText1, U"DESTROY WINDOW ", 15))forget(thePauseForm);
 		if (wasBackgrounding) praat_background ();
 		/* BUG: should also restore praatP. editor. */
 		thePauseForm = NULL;   // undangle
 		thePauseFormRadio = NULL;   // undangle
 		if (thePauseForm_clicked == -1) {
-			Interpreter_stop (interpreter);
-			Melder_throw (U"You interrupted the script.");
+			thePauseForm_clicked = cancelContinueButton;
+			//Interpreter_stop (interpreter);
+			//Melder_throw (U"You interrupted the script.");
 			//Melder_flushError ();
 			//Melder_clearError ();
 		} else {
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.linux.alsa demo_praat_sources/makefiles/makefile.defs.linux.alsa
--- praat_sources/makefiles/makefile.defs.linux.alsa	2011-09-12 10:17:12.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.linux.alsa	2011-09-12 13:28:59.000000000 +0200
@@ -7,7 +7,7 @@ CC = gcc -std=gnu99
 
 CXX = g++ -std=c++11
 
-CFLAGS = -DUNIX -Dlinux -DALSA -D_FILE_OFFSET_BITS=64 `pkg-config --cflags gtk+-2.0` -Werror=missing-prototypes -Werror=implicit -Wreturn-type -Wunused -Wunused-parameter -Wuninitialized -O1 -g1 -pthread
+CFLAGS = -DUNIX -Dlinux -DALSA -D_FILE_OFFSET_BITS=64 `pkg-config --cflags gtk+-2.0` -Werror=missing-prototypes -Werror=implicit -Wreturn-type -Wunused -Wunused-parameter -Wuninitialized -DDEMOAPP -O1 -g1 -pthread
 
 CXXFLAGS = $(CFLAGS) -Wshadow
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.mingw32 demo_praat_sources/makefiles/makefile.defs.mingw32
--- praat_sources/makefiles/makefile.defs.mingw32	2011-09-19 12:26:39.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.mingw32	2011-09-19 15:33:55.000000000 +0200
@@ -7,7 +7,7 @@ CC = /mingw32/bin/gcc -std=gnu99 -isyste
 
 CXX = /mingw32/bin/g++ -std=c++0x -isystem /mingw32/include/c++/4.7.0 -isystem /mingw32/include -Wshadow
 
-CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread
+CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread -DDEMOAPP
 
 CXXFLAGS = $(CFLAGS)
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.mingw64 demo_praat_sources/makefiles/makefile.defs.mingw64--- praat_sources/makefiles/makefile.defs.mingw32	2011-09-19 12:26:39.000000000 +0200
--- praat_sources/makefiles/makefile.defs.mingw64	2011-09-19 12:26:39.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.mingw64	2011-09-19 15:33:37.000000000 +0200
@@ -7,7 +7,7 @@ CC = /mingw64/bin/gcc -std=gnu99 -isyste
 
 CXX = /mingw64/bin/g++ -std=c++0x -isystem /mingw64/include/c++/4.7.0 -isystem /mingw64/include -Wshadow -m64
 
-CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread
+CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread -DDEMOAPP
 
 CXXFLAGS = $(CFLAGS)
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/dwtools/Intensity_extensions.cpp demo_praat_sources/dwtools/Intensity_extensions.cpp
--- praat_sources/dwtools/Intensity_extensions.cpp	2011-09-20 11:40:46.000000000 +0200
+++ demo_praat_sources/dwtools/Intensity_extensions.cpp	2011-09-28 11:45:46.000000000 +0200
@@ -67,8 +67,8 @@ TextGrid Intensity_to_TextGrid_detectSil
 		Vector_getMinimumAndX (me, 0, 0, 1, NUM_PEAK_INTERPOLATE_PARABOLIC, &intensity_min_db, &xOfMinimum);
 		double intensity_dbRange = intensity_max_db - intensity_min_db;
 
-		if (intensity_dbRange < 10) Melder_warning (U"The loudest and softest part in your sound only differ by ",
-			        intensity_dbRange, U" dB.");
+		/*if (intensity_dbRange < 10) Melder_warning (U"The loudest and softest part in your sound only differ by ",
+			        intensity_dbRange, U" dB.");*/
 
 		double intensityThreshold = intensity_max_db - fabs (silenceThreshold_dB);
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat_objectMenus.cpp demo_praat_sources/sys/praat_objectMenus.cpp
--- praat_sources/sys/praat_objectMenus.cpp	2012-02-22 13:04:43.000000000 +0100
+++ demo_praat_sources/sys/praat_objectMenus.cpp	2012-02-22 13:48:24.000000000 +0100
@@ -502,8 +502,10 @@ FORM (SearchManual, U"Search manual", U"
 	TEXTFIELD (U"query", U"")
 	OK2
 DO
+	#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch)
 		Melder_throw (U"Cannot view a manual from batch.");
+	#endif
 	Manual manPage = Manual_create (U"Intro", theCurrentPraatApplication -> manPages, false);
 	Manual_search (manPage, GET_STRING (U"query"));
 END2 }
@@ -514,8 +516,10 @@ FORM (GoToManualPage, U"Go to manual pag
 	LIST (U"Page", numberOfPages, pages, 1)}
 	OK2
 DO
+	#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch)
 		Melder_throw (U"Cannot view a manual from batch.");
+	#endif
 	Manual manPage = Manual_create (U"Intro", theCurrentPraatApplication -> manPages, false);
 	HyperPage_goToPage_i (manPage, GET_INTEGER (U"Page"));
 END2 }
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/GuiMenu.cpp demo_praat_sources/sys/GuiMenu.cpp
--- praat_sources/sys/GuiMenu.cpp	2012-09-27 09:10:50.000000000 +0200
+++ demo_praat_sources/sys/GuiMenu.cpp	2012-10-25 09:55:23.000000000 +0200
@@ -367,7 +367,9 @@ GuiMenu GuiMenu_createInWindow (GuiWindo
 
 		}
 	#elif motif
+		#if not ( defined( DEMOAPP ) && defined( macintosh ) )
 		if (window == NULL) {
+		#endif
 			my d_xmMenuTitle = XmCreateCascadeButton (theGuiTopMenuBar, Melder_peek32to8 (title), NULL, 0);
 			if (str32equ (title, U"Help"))
 				XtVaSetValues (theGuiTopMenuBar, XmNmenuHelpWidget, my d_xmMenuTitle, NULL);
@@ -376,6 +378,7 @@ GuiMenu GuiMenu_createInWindow (GuiWindo
 				XtSetSensitive (my d_xmMenuTitle, False);
 			XtVaSetValues (my d_xmMenuTitle, XmNsubMenuId, my d_widget, NULL);
 			XtManageChild (my d_xmMenuTitle);
+		#if not ( defined( DEMOAPP ) && defined( macintosh ) )
 		} else {
 			my d_xmMenuTitle = XmCreateCascadeButton (window -> d_xmMenuBar, Melder_peek32to8 (title), NULL, 0);
 			if (str32equ (title, U"Help"))
@@ -386,6 +389,7 @@ GuiMenu GuiMenu_createInWindow (GuiWindo
 			XtVaSetValues (my d_xmMenuTitle, XmNsubMenuId, my d_widget, NULL);
 			XtManageChild (my d_xmMenuTitle);
 		}
+		#endif
 		_GuiObject_setUserData (my d_widget, me);
 	#endif
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/DemoEditor.cpp demo_praat_sources/sys/DemoEditor.cpp
--- praat_sources/sys/DemoEditor.cpp	2014-05-09 08:59:02.000000000 +0200
+++ demo_praat_sources/sys/DemoEditor.cpp	2014-05-16 10:30:28.000000000 +0200
@@ -249,6 +249,7 @@ void Demo_waitForInput (Interpreter inte
 	if (theDemoEditor -> userWantsToClose) {
 		Interpreter_stop (interpreter);
 		forget (theDemoEditor);
+		exit (0);
 		Melder_throw (U"You interrupted the script.");
 	}
 }
