diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/melder_audio.cpp demo_praat_sources/sys/melder_audio.cpp
--- praat_sources/sys/melder_audio.cpp	2014-12-29 08:24:04.000000000 +0100
+++ demo_praat_sources/sys/melder_audio.cpp	2015-05-12 09:48:06.000000000 +0200
# Allow playback in the background
@@ -1019,10 +1019,14 @@ void MelderAudio_play16 (int16_t *buffer
 	my numberOfChannels = numberOfChannels;
 	my callback = playCallback;
 	my closure = playClosure;
+	#ifdef DEMOAPP
+		my asynchronicity = kMelder_asynchronicityLevel_ASYNCHRONOUS;
+	#else
 	my asynchronicity =
 		Melder_batch ? kMelder_asynchronicityLevel_SYNCHRONOUS :
 		(Melder_backgrounding && ! Melder_asynchronous) ? kMelder_asynchronicityLevel_INTERRUPTABLE :
 		kMelder_asynchronicityLevel_ASYNCHRONOUS;
+	#endif
 	if (my asynchronicity > preferences. maximumAsynchronicity)
 		my asynchronicity = preferences. maximumAsynchronicity;
 	trace (U"asynchronicity ", my asynchronicity);
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat_script.cpp demo_praat_sources/sys/praat_script.cpp
--- praat_sources/sys/praat_script.cpp	2011-09-02 13:53:33.000000000 +0200
+++ demo_praat_sources/sys/praat_script.cpp	2011-10-28 07:54:25.000000000 +0200
# In case of an error, terminate the program
@@ -602,6 +602,7 @@ void praat_executeScriptFromText (wchar 
 		autostring32 string = Melder_dup (text);   // copy, because Interpreter will change it (UGLY)
 		Interpreter_run (interpreter.get(), string.peek());
 	} catch (MelderError) {
+		exit (0);
 		Melder_throw (U"Script not completed.");
 	}
 }
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/Graphics_text.cpp demo_praat_sources/sys/Graphics_text.cpp
--- praat_sources/sys/Graphics_text.cpp	2011-09-29 12:27:01.000000000 +0200
+++ demo_praat_sources/sys/Graphics_text.cpp	2011-10-27 09:14:57.000000000 +0200
# Supress the annoying message about the phonetic font not being installed
@@ -105,7 +105,7 @@ static HFONT loadFont (GraphicsScreen me
 			ipaInited = true;
 			if (! charisAvailable && ! doulosAvailable) {
 				/* BUG: The next warning may cause reentry of drawing (on window exposure) and lead to crash. Some code must be non-reentrant !! */
-				Melder_warning (U"The phonetic font is not available.\nSeveral characters may not look correct.\nSee www.praat.org");
+				//Melder_warning (U"The phonetic font is not available.\nSeveral characters may not look correct.\nSee www.praat.org");
 			}
 		}
 		wcscpy (spec. lfFaceName,
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat.cpp demo_praat_sources/sys/praat.cpp
--- praat_sources/sys/praat.cpp	2016-01-11 08:45:44.000000000 +0100
+++ demo_praat_sources/sys/praat.cpp	2016-01-27 15:29:23.000000000 +0100
# When running thePraatStandAloneScriptText, block the Melder_batch flag
@@ -1136,7 +1136,9 @@ void praat_init (const char32 *title, in
 	}
 	//Melder_casual (U"Script file name <<", theCurrentPraatApplication -> batchName.string, U">>");
 
+	#if ! ( defined (_WIN32) && defined (DEMOAPP) )
 	Melder_batch |= !! thePraatStandAloneScriptText;
+	#endif
 
 	/*
 	 * Running the Praat shell from the command line:
# Change the name of the Unix preference dir
@@ -1188,7 +1190,7 @@ void praat_init (const char32 *title, in
 		 */
 		char32 name [256];
 		#if defined (UNIX)
-			Melder_sprint (name,256, U".", programName, U"-dir");   // for example .myProg-dir
+			Melder_sprint (name,256, U".", programName, U"");   // for example .myProg-dir
 		#elif defined (macintosh)
 			Melder_sprint (name,256, praatP.title, U" Prefs");   // for example MyProg Prefs
 		#elif defined (_WIN32)
# Something in Linux (GTK)? Do not know what?
@@ -1272,6 +1274,12 @@ void praat_init (const char32 *title, in
 
 	GuiWindow raam = nullptr;
 	if (Melder_batch) {
+		#if defined (DEMOAPP) && gtk
+			Machine_initLookAndFeel (argc, argv);
+			char32 objectWindowTitle [100];
+			Melder_sprint (objectWindowTitle,100, praatP.title, U"");
+			GuiWindow_create (0, 0, WINDOW_WIDTH, WINDOW_HEIGHT, 1, 1, objectWindowTitle, gui_cb_quit, NULL, 0);
+		#endif
 		MelderString_empty (& theCurrentPraatApplication -> batchName);
 		for (int i = praatP.argumentNumber - 1; i < argc; i ++) {
 			if (i >= praatP.argumentNumber) MelderString_append (& theCurrentPraatApplication -> batchName, U" ");
# Supress opening the object window, initialize praat under Mac 64 bit (cocoa) Needed for 10.7
@@ -1313,10 +1321,32 @@ void praat_init (const char32 *title, in
 		#if motif
 			GuiApp_setApplicationShell (theCurrentPraatApplication -> topShell -> d_xmShell);
 		#endif
+#ifndef DEMOAPP
 		trace (U"before objects window shows locale ", Melder_peek8to32 (setlocale (LC_ALL, nullptr)));
 		GuiThing_show (raam);
 		trace (U"after objects window shows locale ", Melder_peek8to32 (setlocale (LC_ALL, nullptr)));
+#endif
 	}
+
+	#if cocoa
+	{
+		trace (U"starting the application");
+		Machine_initLookAndFeel (argc, argv);
+		/*
+		 * Start the application.
+		 */
+		//[NSApplication sharedApplication];
+		[GuiCocoaApplication sharedApplication];
+
+		trace (U"creating and installing the Objects window");
+		char32 objectWindowTitle [100];
+		double x, y;
+		trace (U"locale ", Melder_peek8to32 (setlocale (LC_ALL, nullptr)));
+		Gui_getWindowPositioningBounds (& x, & y, nullptr, nullptr);
+		trace (U"locale ", Melder_peek8to32 (setlocale (LC_ALL, nullptr)));
+	}
+	#endif
+
 	Thing_recognizeClassesByName (classCollection, classStrings, classManPages, classStringSet, nullptr);
 	Thing_recognizeClassByOtherName (classStringSet, U"SortedSetOfString");
 	if (Melder_batch) {
# Seems to speed startup, but not sure whether it is necessary
@@ -1326,6 +1356,9 @@ void praat_init (const char32 *title, in
 		trace (U"adding fixed buttons without GUI");
 		praat_addFixedButtons (nullptr);
 	} else {
+		#ifdef DEMOAPP
+			Melder_backgrounding = true;
+		#endif
 
 		#ifdef macintosh
 			AEInstallEventHandler (758934755, 0, (AEEventHandlerProcPtr) (mac_processSignal8), 0, false);   // for receiving sendpraat
# Supress opening the object window
@@ -1347,7 +1380,9 @@ void praat_init (const char32 *title, in
 		trace (U"creating the dynamic menu in the Objects window");
 		praat_actions_createDynamicMenu (raam);
 		trace (U"showing the Objects window");
+#ifndef DEMOAPP
 		GuiThing_show (raam);
+#endif
 	//Melder_fatal (U"stop");
 		#if defined (UNIX) && ! defined (NO_GRAPHICS)
 			try {
# Initialize a picture window, but close it immediately
@@ -1378,7 +1413,12 @@ void praat_init (const char32 *title, in
 
 	trace (U"creating the Picture window");
 	trace (U"before picture window shows: locale is ", Melder_peek8to32 (setlocale (LC_ALL, nullptr)));
-	if (! praatP.dontUsePictureWindow) praat_picture_init ();
+	#ifdef DEMOAPP
+		praat_picture_init ();
+		praat_dontUsePictureWindow ();
+	#else
+		if (! praatP.dontUsePictureWindow) praat_picture_init ();
+	#endif
 	trace (U"after picture window shows: locale is ", Melder_peek8to32 (setlocale (LC_ALL, nullptr)));
 
 	if (unknownCommandLineOption) {
# No idea why this is needed?
@@ -1541,7 +1581,7 @@ void praat_run () {
 			Melder_fatal (U"The number 1000000000000 is mistakenly written on this machine as ", dummy, U".");
 	}
 	{ uint32_t dummy = 0xffffffff;
-		Melder_assert ((int64) dummy == 4294967295LL);
+		Melder_assert ((int64_t) dummy == 4294967295LL);
 		Melder_assert (str32equ (Melder_integer (dummy), U"4294967295"));
 		Melder_assert (double (dummy) == 4294967295.0);
 	}
# Next chunk is needed to start the Demo window under Windows (motif)
@@ -1653,10 +1693,20 @@ void praat_run () {
 			trace (U"locale is ", Melder_peek8to32 (setlocale (LC_ALL, nullptr)));
 			gtk_main ();
 		#elif motif
-			for (;;) {
-				XEvent event;
-				GuiNextEvent (& event);
-				XtDispatchEvent (& event);
+			if (thePraatStandAloneScriptText) {
+				try {
+					praat_executeScriptFromText (thePraatStandAloneScriptText);
+					praat_exit (0);
+				} catch (MelderError) {
+					Melder_flushError (praatP.title, U": stand-alone script session interrupted.");
+					praat_exit (-1);
+				}
+			} else {
+				for (;;) {
+					XEvent event;
+					GuiNextEvent (& event);
+					XtDispatchEvent (& event);
+				} 
 			}
 		#elif cocoa
 			[NSApp run];
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat_picture.cpp demo_praat_sources/sys/praat_picture.cpp
--- praat_sources/sys/praat_picture.cpp	2012-09-27 09:10:50.000000000 +0200
+++ demo_praat_sources/sys/praat_picture.cpp	2012-10-15 13:47:43.000000000 +0200
# Supress picture window
@@ -1595,8 +1595,10 @@ void praat_picture_open (void) {
 		#if gtk
 			gtk_window_present (GTK_WINDOW (dialog -> d_gtkWindow));
 		#elif motif
+#ifndef DEMOAPP
 			XtMapWidget (dialog -> d_xmShell);
 			XMapRaised (XtDisplay (dialog -> d_xmShell), XtWindow (dialog -> d_xmShell));
+#endif
 		#elif cocoa
 			GuiThing_show (dialog);
 		#endif
@@ -1888,7 +1890,9 @@ void praat_picture_init (void) {
 		width = height = resolution * 12;
 		scrollWindow = GuiScrolledWindow_createShown (dialog, margin, 0, Machine_getMenuBarHeight () + margin, 0, 1, 1, 0);
 		drawingArea = GuiDrawingArea_createShown (scrollWindow, width, height, nullptr, nullptr, nullptr, nullptr, nullptr, 0);
+#ifndef DEMOAPP
 		GuiThing_show (dialog);
+#endif
 	}
 
 	// TODO: Paul: deze moet VOOR de update functies anders krijgen die void_me 0x0
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/fon/praat_Sound.cpp demo_praat_sources/fon/praat_Sound.cpp
--- praat_sources/fon/praat_Sound.cpp	2017-06-08 11:26:20.000000000 +0200
+++ demo_praat_sources/fon/praat_Sound.cpp	2017-06-08 12:22:45.000000000 +0200
# View and edit Long/Sounds in a DemoApp (when Melder_back = True)
@@ -157,7 +157,9 @@ DO
 }
 
 DIRECT (WINDOW_LongSound_view) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a LongSound from batch.");
+#endif
 	LOOP {
 		iam (LongSound);
 		autoSoundEditor editor = SoundEditor_create (ID_AND_FULL_NAME, me);
@@ -671,7 +673,9 @@ static void cb_SoundEditor_publication (
 	}
 }
 DIRECT (WINDOW_Sound_viewAndEdit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a Sound from batch.");
+#endif
 	LOOP {
 		iam (Sound);
 		autoSoundEditor editor = SoundEditor_create (ID_AND_FULL_NAME, me);
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/Ui.cpp demo_praat_sources/sys/Ui.cpp
--- praat_sources/sys/Ui.cpp	2014-01-27 09:02:44.000000000 +0100
+++ demo_praat_sources/sys/Ui.cpp	2014-03-27 12:22:58.000000000 +0100
# Increase size of Pause Forms
@@ -841,12 +841,16 @@ void UiForm_finish (I) {
 void UiForm_finish (UiForm me) {
 	if (! my d_dialogParent && ! my isPauseForm) return;
 
+	int numberOfContinueButtons = my numberOfContinueButtons;
 	int size = my numberOfFields;
 	int dialogHeight = 0, x = Gui_LEFT_DIALOG_SPACING, y;
 	int textFieldHeight = Gui_TEXTFIELD_HEIGHT;
-	int dialogWidth = 520, dialogCentre = dialogWidth / 2, fieldX = dialogCentre + Gui_LABEL_SPACING / 2;
+	int dialogWidth = REVERT_BUTTON_WIDTH + STOP_BUTTON_WIDTH + 2*Gui_LABEL_SPACING + numberOfContinueButtons*(STANDARDS_BUTTON_WIDTH + Gui_LABEL_SPACING), dialogCentre = dialogWidth / 2, fieldX = dialogCentre + Gui_LABEL_SPACING / 2;
 	int labelWidth = fieldX - Gui_LABEL_SPACING - x, fieldWidth = labelWidth, halfFieldWidth = fieldWidth / 2 - 6;
 
+	/* Ensure that the dialogs are wide enough */
+	if (dialogWidth<520) dialogWidth = 520;
+	
 	GuiForm form;
 
 	/*
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/UiPause.cpp demo_praat_sources/sys/UiPause.cpp
--- praat_sources/sys/UiPause.cpp	2012-09-27 09:10:50.000000000 +0200
+++ demo_praat_sources/sys/UiPause.cpp	2012-09-27 13:25:18.000000000 +0200
# Catch close buttons of Pause window and prevent terminating the script
@@ -143,6 +143,7 @@ int UiPause_end (int numberOfContinueBut
 	{// scope
 		autoMelderSaveDefaultDir saveDir;
 		thePauseForm_clicked = 0;
+		if(str32nequ(continueText1, U"DESTROY WINDOW ", 15))thePauseForm_clicked = 1;
 		Melder_assert (theEventLoopDepth == 0);
 		theEventLoopDepth ++;
 		try {
@@ -166,24 +167,28 @@ int UiPause_end (int numberOfContinueBut
 					[pool release];
 				} while (! thePauseForm_clicked);
 			#elif motif
+				if(str32nequ(continueText1, U"DESTROY WINDOW ", 15) == 0){
 				do {
 					XEvent event;
 					GuiNextEvent (& event);
 					XtDispatchEvent (& event);
 				} while (! thePauseForm_clicked);
+				};
 			#endif
 		} catch (MelderError) {
 			Melder_flushError (U"An error made it to the outer level in a pause window; should not occur! Please write to paul.boersma@uva.nl");
 		}
 		theEventLoopDepth --;
 	}
+		if(str32nequ(continueText1, U"DESTROY WINDOW ", 15))forget(thePauseForm);
 	if (wasBackgrounding) praat_background ();
 	/* BUG: should also restore praatP. editor. */
 	thePauseForm = nullptr;   // undangle
 	thePauseFormRadio = nullptr;   // undangle
 	if (thePauseForm_clicked == -1) {
-		Interpreter_stop (interpreter);
-		Melder_throw (U"You interrupted the script.");
+		thePauseForm_clicked = cancelContinueButton;
+		//Interpreter_stop (interpreter);
+		//Melder_throw (U"You interrupted the script.");
 		//Melder_flushError ();
 		//Melder_clearError ();
 	} else {
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/dwtools/Intensity_extensions.cpp demo_praat_sources/dwtools/Intensity_extensions.cpp
--- praat_sources/dwtools/Intensity_extensions.cpp	2011-09-20 11:40:46.000000000 +0200
+++ demo_praat_sources/dwtools/Intensity_extensions.cpp	2011-09-28 11:45:46.000000000 +0200
@@ -65,9 +65,9 @@ TextGrid Intensity_to_TextGrid_detectSil
 		Vector_getMinimumAndX (me, 0, 0, 1, NUM_PEAK_INTERPOLATE_PARABOLIC, &intensity_min_db, &xOfMinimum);
 		double intensity_dbRange = intensity_max_db - intensity_min_db;
 
-		if (intensity_dbRange < 10) {
-			Melder_warning (U"The loudest and softest part in your sound only differ by ", intensity_dbRange, U" dB.");
-		}
+		/*if (intensity_dbRange < 10) {
+			Melder_warning (U"The loudest and softest part in your sound only differ by ", intensity_dbRange, U" dB.");
+		}*/
 		double intensityThreshold = intensity_max_db - fabs (silenceThreshold_dB);
 
 		if (minSilenceDuration > duration || intensityThreshold < intensity_min_db) {
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/praat_objectMenus.cpp demo_praat_sources/sys/praat_objectMenus.cpp
--- praat_sources/sys/praat_objectMenus.cpp	2012-02-22 13:04:43.000000000 +0100
+++ demo_praat_sources/sys/praat_objectMenus.cpp	2012-02-22 13:48:24.000000000 +0100
# Allow the man pages
@@ -511,8 +511,10 @@ FORM (SearchManual, U"Search manual", U"
 	TEXTFIELD4 (query, U"query", U"")
 	OK
 DO
+	#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch)
 		Melder_throw (U"Cannot view a manual from batch.");
+	#endif
 	autoManual manual = Manual_create (U"Intro", theCurrentPraatApplication -> manPages, false);
 	Manual_search (manual.get(), query);
 	manual.releaseToUser();
@@ -524,8 +526,10 @@ FORM (GoToManualPage, U"Go to manual pag
 	LIST (U"Page", numberOfPages, pages, 1)
 	OK
 DO
+	#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch)
 		Melder_throw (U"Cannot view a manual from batch.");
+	#endif
 	autoManual manual = Manual_create (U"Intro", theCurrentPraatApplication -> manPages, false);
 	HyperPage_goToPage_i (manual.get(), GET_INTEGER (U"Page"));
 	manual.releaseToUser();
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/GuiMenu.cpp demo_praat_sources/sys/GuiMenu.cpp
--- praat_sources/sys/GuiMenu.cpp	2017-06-08 11:26:20.000000000 +0200
+++ demo_praat_sources/sys/GuiMenu.cpp	2017-06-09 15:29:42.000000000 +0200
@@ -313,7 +313,9 @@ GuiMenu GuiMenu_createInWindow (GuiWindo
 		gtk_menu_item_set_submenu (GTK_MENU_ITEM (my d_gtkMenuTitle), GTK_WIDGET (my d_widget));
 		_GuiObject_setUserData (my d_widget, me.get());
 	#elif motif
+		#if not ( defined( DEMOAPP ) && defined( macintosh ) )
 		Melder_assert (window);
+		#endif
 		my d_xmMenuTitle = XmCreateCascadeButton (window -> d_xmMenuBar, Melder_peek32to8 (title), nullptr, 0);
 		if (str32equ (title, U"Help"))
 			XtVaSetValues (window -> d_xmMenuBar, XmNmenuHelpWidget, my d_xmMenuTitle, nullptr);
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/sys/DemoEditor.cpp demo_praat_sources/sys/DemoEditor.cpp
--- praat_sources/sys/DemoEditor.cpp	2014-05-09 08:59:02.000000000 +0200
+++ demo_praat_sources/sys/DemoEditor.cpp	2014-05-16 10:30:28.000000000 +0200
# Close the script and stop Praat. If you do not stop praat, you retain Zombie Praat processes under Windows
@@ -266,6 +266,7 @@ void Demo_waitForInput (Interpreter inte
 	if (theReferenceToTheOnlyDemoEditor -> userWantsToClose) {
 		Interpreter_stop (interpreter);
 		forget (theReferenceToTheOnlyDemoEditor);
+		exit (0);
 		Melder_throw (U"You interrupted the script.");
 	}
 }
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/stat/praat_Stat.cpp demo_praat_sources/stat/praat_Stat.cpp
--- praat_sources/stat/praat_Stat.cpp	2017-06-08 11:26:20.000000000 +0200
+++ demo_praat_sources/stat/praat_Stat.cpp	2017-06-09 15:14:12.000000000 +0200
# Allow to view tables when in a Demo app
@@ -291,7 +291,9 @@ DIRECT (HELP_StatisticsTutorial) {
 // MARK: View & Edit
 
 DIRECT (WINDOW_Table_viewAndEdit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot edit a Table from batch.");
+#endif
 	LOOP {
 		iam_LOOP (Table);
 		autoTableEditor editor = TableEditor_create (ID_AND_FULL_NAME, me);
--- praat_sources/stat/praat_Stat.cpp	2011-10-01 13:32:45.000000000 +0200
+++ demo_praat_sources/stat/praat_Stat.cpp	2011-11-18 08:55:32.000000000 +0100
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/fon/praat_TextGrid_init.cpp demo_praat_sources/fon/praat_TextGrid_init.cpp
--- praat_sources/fon/praat_TextGrid_init.cpp	2017-06-08 11:26:20.000000000 +0200
+++ demo_praat_sources/fon/praat_TextGrid_init.cpp	2017-06-09 15:49:13.000000000 +0200
# View and edit TextGrids in a DemoApp (when Melder_back = True)
@@ -703,7 +703,9 @@ static void cb_TextGridEditor_publicatio
 	}
 }
 DIRECT (WINDOW_TextGrid_viewAndEdit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	Sound sound = nullptr;
 	LOOP {
 		if (CLASS == classSound) sound = (Sound) OBJECT;   // may stay null
@@ -721,7 +723,9 @@ FORM (WINDOW_TextGrid_viewAndEditWithCal
 	SENTENCE4 (callbackText, U"Callback text", U"r1")
 	OK
 DO
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	Sound sound = nullptr;
 	LOOP {
 		if (CLASS == classSound) sound = (Sound) OBJECT;   // may stay null
@@ -736,7 +740,9 @@ DO
 END }
 
 DIRECT (WINDOW_TextGrid_LongSound_viewAndEdit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	LongSound longSound = nullptr;
 	int ilongSound = 0;
 	LOOP {
@@ -753,7 +759,9 @@ DIRECT (WINDOW_TextGrid_LongSound_viewAn
 END }
 
 DIRECT (WINDOW_TextGrid_SpellingChecker_viewAndEdit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	SpellingChecker spellingChecker = nullptr;
 	int ispellingChecker = 0;
 	Sound sound = nullptr;
@@ -771,7 +779,9 @@ DIRECT (WINDOW_TextGrid_SpellingChecker_
 END }
 
 DIRECT (WINDOW_TextGrid_LongSound_SpellingChecker_viewAndEdit) {
+#ifndef DEMOAPP
 	if (theCurrentPraatApplication -> batch) Melder_throw (U"Cannot view or edit a TextGrid from batch.");
+#endif
 	LongSound longSound = nullptr;
 	SpellingChecker spellingChecker = nullptr;
 	int ilongSound = 0, ispellingChecker = 0;
#
# Add DEMOAPP flag to compiler
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.linux.alsa demo_praat_sources/makefiles/makefile.defs.linux.alsa
--- praat_sources/makefiles/makefile.defs.linux.alsa	2011-09-12 10:17:12.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.linux.alsa	2011-09-12 13:28:59.000000000 +0200
@@ -7,7 +7,7 @@ CC = gcc -std=gnu99
 
 CXX = g++ -std=c++11
 
-CFLAGS = -DUNIX -Dlinux -DALSA -D_FILE_OFFSET_BITS=64 `pkg-config --cflags gtk+-2.0` -Werror=missing-prototypes -Werror=implicit -Wreturn-type -Wunused -Wunused-parameter -Wuninitialized -O1 -g1 -pthread
+CFLAGS = -DUNIX -Dlinux -DALSA -D_FILE_OFFSET_BITS=64 `pkg-config --cflags gtk+-2.0` -Werror=missing-prototypes -Werror=implicit -Wreturn-type -Wunused -Wunused-parameter -Wuninitialized -DDEMOAPP -O1 -g1 -pthread
 
 CXXFLAGS = $(CFLAGS) -Wshadow
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.darmin32 demo_praat_sources/makefiles/makefile.defs.darmin32
--- praat_sources/makefiles/makefile.defs.darmin32	2011-09-19 12:26:39.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.darmin32	2011-09-19 15:33:55.000000000 +0200
@@ -7,7 +7,7 @@ CC = /mingw32/bin/gcc -std=gnu99 -isyste
 
 CXX = /mingw32/bin/g++ -std=c++0x -isystem /mingw32/include/c++/4.7.0 -isystem /mingw32/include -Wshadow
 
-CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread
+CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread -DDEMOAPP
 
 CXXFLAGS = $(CFLAGS)
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.darmin64 demo_praat_sources/makefiles/makefile.defs.darmin64
--- praat_sources/makefiles/makefile.defs.darmin64	2011-09-19 12:26:39.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.darmin64	2011-09-19 15:33:37.000000000 +0200
@@ -7,7 +7,7 @@ CC = /mingw64/bin/gcc -std=gnu99 -isyste
 
 CXX = /mingw64/bin/g++ -std=c++0x -isystem /mingw64/include/c++/4.7.0 -isystem /mingw64/include -Wshadow -m64
 
-CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread
+CFLAGS = -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0500 -DUNICODE -D_FILE_OFFSET_BITS=64 -Dmain=wingwmain -O1 -pthread -DDEMOAPP
 
 CXXFLAGS = $(CFLAGS)
 
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.mingw32 demo_praat_sources/makefiles/makefile.defs.mingw32
--- praat_sources/makefiles/makefile.defs.mingw32	2015-09-11 08:38:17.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.mingw32	2015-09-15 15:45:42.000000000 +0200
@@ -9,7 +9,7 @@ CC = $(PREFIX)gcc -std=gnu99
 
 CXX = $(PREFIX)g++ -std=gnu++11 -Wshadow
 
-CFLAGS = -municode -D_FILE_OFFSET_BITS=64 -O3
+CFLAGS = -municode -D_FILE_OFFSET_BITS=64 -O3 -DDEMOAPP
 # Probably implicit: -m32 -mwin32 -march=i686 -mtune=generic
 
 CXXFLAGS = $(CFLAGS)
diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/makefiles/makefile.defs.mingw64 demo_praat_sources/makefiles/makefile.defs.mingw64
--- praat_sources/makefiles/makefile.defs.mingw64	2015-09-11 08:38:17.000000000 +0200
+++ demo_praat_sources/makefiles/makefile.defs.mingw64	2015-09-15 15:48:18.000000000 +0200
@@ -9,7 +9,7 @@ CC = $(PREFIX)gcc -std=gnu99
 
 CXX = $(PREFIX)g++ -std=gnu++11 -Wshadow
 
-CFLAGS = -municode -D_FILE_OFFSET_BITS=64 -O3
+CFLAGS = -municode -D_FILE_OFFSET_BITS=64 -O3 -DDEMOAPP
 # Probably implicit: -m64 -mwin32 -march=x86_64 -mtune=generic
 
 CXXFLAGS = $(CFLAGS)
