diff -Naurp -x '*.[oah]' -x '.git*' -x praat praat_sources/main/main_Praat.cpp demo_praat_sources/main/main_Praat.cpp
--- praat_sources/main/main_Praat.cpp	2012-09-27 10:10:58.000000000 +0200
+++ demo_praat_sources/main/main_Praat.cpp	2012-09-27 10:13:36.000000000 +0200
@@ -44,12 +44,14 @@ static void logo (Graphics g) {
 	Graphics_text (g, 0.5, 0.16, L"Copyright \\co 1992-" xstr(PRAAT_YEAR) " by Paul Boersma and David Weenink");
 }
 
+#include "demopraatapplication.h"
 int main (int argc, char *argv []) {
 	#if cocoa
 		Melder_setTracing (true);
 	#endif
 	praat_setLogo (130, 80, logo);
-	praat_init ("Praat", argc, argv);
+	praat_setStandAloneScriptText (myDemoScript);
+	praat_init ("TEanalysis", argc, argv);
 	INCLUDE_LIBRARY (praat_uvafon_init)
 	INCLUDE_LIBRARY (praat_contrib_Ola_KNN_init)
 	praat_run ();
