--- praat_sources/main/main_Praat.cpp	2013-06-26 09:06:04.000000000 +0200
+++ demo_praat_sources/main/main_Praat.cpp	2013-06-28 14:47:39.000000000 +0200
@@ -44,10 +44,12 @@ static void logo (Graphics g) {
 	Graphics_text (g, 0.5, 0.16, L"Copyright \\co 1992-" xstr(PRAAT_YEAR) " by Paul Boersma and David Weenink");
 }
 
+#include "demopraatapplication.h"
 int main (int argc, char *argv []) {
 	try {
 		praat_setLogo (130, 80, logo);
-		praat_init ("Praat", argc, argv);
+		praat_setStandAloneScriptText (myDemoScript);
+		praat_init ("TEVA", argc, argv);
 		INCLUDE_LIBRARY (praat_uvafon_init)
 		INCLUDE_LIBRARY (praat_contrib_Ola_KNN_init)
 		praat_run ();
